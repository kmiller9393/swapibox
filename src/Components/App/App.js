import React, { Component } from 'react';
import { CardContainer } from '../CardContainer/CardContainer';
import Favorites from '../Favorites/Favorites';
import { Navbar } from '../Navbar/Navbar';
import Sidebar from '../Sidebar/Sidebar';
import './App.css';
import {
  peopleDataFetcher,
  planetDataFetcher,
  vehicleDataFetcher
} from '../../apiCalls';

export default class App extends Component {
  constructor() {
    super();
    this.state = {
      people: [],
      planets: [],
      vehicles: [],
      currentView: '',
      isLoading: false
    };
  }

  setContainerView = async endpoint => {
    switch (endpoint) {
      case 'people':
        const people = await peopleDataFetcher('people');
        this.setState({ people, currentView: 'people' });
        break;
      case 'planets':
        const planets = await planetDataFetcher('planets');
        this.setState({ planets, currentView: 'planets' });
        break;
      case 'vehicles':
        const vehicles = await vehicleDataFetcher('vehicles');
        this.setState({ vehicles, currentView: 'vehicles' });
        break;
      default:
      //loading screen code here;
    }
  };

  render() {
    return (
      <div className="container">
        <div className="header">
          <h1 className="header-title">SwapiBox</h1>
        </div>
        <Navbar setContainerView={this.setContainerView} />
        <Sidebar />
        {this.state.currentView && (
          <CardContainer selectedGroup={this.state[this.state.currentView]} />
        )}
        <Favorites />
      </div>
    );
  }
}
