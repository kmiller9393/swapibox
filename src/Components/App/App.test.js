import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import {
  peopleDataFetcher,
  planetDataFetcher,
  vehicleDataFetcher
} from '../../apiCalls';
import { mockAppPerson, mockAppPlanets, mockAppVehicles } from '../../mockData';
import { shallow } from 'enzyme';

describe('App', () => {
  let wrapper;
  let mockPerson;
  let mockPerson2;

  beforeEach(() => {
    wrapper = shallow(<App />);
    mockPerson = [
      {
        name: 'Luke Skywalker',
        favorite: false
      }
    ];
    mockPerson2 = [
      {
        name: 'Luke Skywalker',
        favorite: true
      }
    ];
  });

  it('matches the snapshot on load', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('matches the snapshot when there are favorites', () => {
    wrapper.setState({ currentView: 'favorites' });
    expect(wrapper).toMatchSnapshot();
  });

  it('matches the snapshot when the currentView has a value other than favorites', () => {
    wrapper.setState({ currentView: 'planets' });
    expect(wrapper).toMatchSnapshot();
  });

  it('favoriteItem puts the current card in favorites', () => {
    wrapper.setState({ currentView: 'people' });
    wrapper.setState({ people: mockPerson });
    wrapper.instance().favoriteItem('Luke Skywalker');
    expect(wrapper.state('favorites')).toEqual(mockPerson2);
  });

  it('favoriteItem should remove an item from favorites if it already exists', () => {
    wrapper.setState({ currentView: 'people' });
    wrapper.setState({ people: mockPerson });
    wrapper.instance().favoriteItem('Luke Skywalker');
    wrapper.instance().favoriteItem('Luke Skywalker');
    expect(wrapper.state('favorites')).toEqual([]);
  });

  it('setContainerView should call peopleDataFetcher when passed in a string of people', async () => {
    await wrapper.instance().setContainerView('people');
    expect(wrapper.state('people')).toEqual(mockAppPerson);
  });

  it('setContainerView should call planetDataFetcher when passed in a string of people', async () => {
    await wrapper.instance().setContainerView('planets');
    expect(wrapper.state('planets')).toEqual(mockAppPlanets);
  });

  it('setContainerView should call vehicleDataFetcher when passed in a string of people', async () => {
    await wrapper.instance().setContainerView('vehicles');
    expect(wrapper.state('vehicles')).toEqual(mockAppVehicles);
  });

  it('setContainerView should change state to currentView of favorites', async () => {
    await wrapper.instance().setContainerView('favorites');
    expect(wrapper.state('currentView')).toEqual('favorites');
  });
});
