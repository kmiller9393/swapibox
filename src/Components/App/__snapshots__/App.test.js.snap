// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches the snapshot on load 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="header"
>
          <h1
                    className="header-title"
          >
                    Swapi-Box
          </h1>
</div>,
        <Navbar
          setContainerView={[Function]}
/>,
        <Sidebar />,
        <div
          className="card-container start"
>
          Select A Category
</div>,
        false,
        "",
        false,
        <Unknown
          numberOfFavorites={0}
          setContainerView={[Function]}
/>,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1
            className="header-title"
>
            Swapi-Box
</h1>,
          "className": "header",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Swapi-Box",
            "className": "header-title",
          },
          "ref": null,
          "rendered": "Swapi-Box",
          "type": "h1",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "setContainerView": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Select A Category",
          "className": "card-container start",
        },
        "ref": null,
        "rendered": "Select A Category",
        "type": "div",
      },
      false,
      "",
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "numberOfFavorites": 0,
          "setContainerView": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="header"
>
            <h1
                        className="header-title"
            >
                        Swapi-Box
            </h1>
</div>,
          <Navbar
            setContainerView={[Function]}
/>,
          <Sidebar />,
          <div
            className="card-container start"
>
            Select A Category
</div>,
          false,
          "",
          false,
          <Unknown
            numberOfFavorites={0}
            setContainerView={[Function]}
/>,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1
              className="header-title"
>
              Swapi-Box
</h1>,
            "className": "header",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Swapi-Box",
              "className": "header-title",
            },
            "ref": null,
            "rendered": "Swapi-Box",
            "type": "h1",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "setContainerView": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Select A Category",
            "className": "card-container start",
          },
          "ref": null,
          "rendered": "Select A Category",
          "type": "div",
        },
        false,
        "",
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "numberOfFavorites": 0,
            "setContainerView": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`App matches the snapshot when the currentView has a value other than favorites 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="header"
>
          <h1
                    className="header-title"
          >
                    Swapi-Box
          </h1>
</div>,
        <Navbar
          setContainerView={[Function]}
/>,
        <Sidebar />,
        false,
        false,
        <Unknown
          favoriteItem={[Function]}
          selectedGroup={Array []}
/>,
        false,
        <Unknown
          numberOfFavorites={0}
          setContainerView={[Function]}
/>,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1
            className="header-title"
>
            Swapi-Box
</h1>,
          "className": "header",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Swapi-Box",
            "className": "header-title",
          },
          "ref": null,
          "rendered": "Swapi-Box",
          "type": "h1",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "setContainerView": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      false,
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "favoriteItem": [Function],
          "selectedGroup": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "numberOfFavorites": 0,
          "setContainerView": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="header"
>
            <h1
                        className="header-title"
            >
                        Swapi-Box
            </h1>
</div>,
          <Navbar
            setContainerView={[Function]}
/>,
          <Sidebar />,
          false,
          false,
          <Unknown
            favoriteItem={[Function]}
            selectedGroup={Array []}
/>,
          false,
          <Unknown
            numberOfFavorites={0}
            setContainerView={[Function]}
/>,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1
              className="header-title"
>
              Swapi-Box
</h1>,
            "className": "header",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Swapi-Box",
              "className": "header-title",
            },
            "ref": null,
            "rendered": "Swapi-Box",
            "type": "h1",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "setContainerView": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        false,
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "favoriteItem": [Function],
            "selectedGroup": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "numberOfFavorites": 0,
            "setContainerView": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`App matches the snapshot when there are favorites 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="header"
>
          <h1
                    className="header-title"
          >
                    Swapi-Box
          </h1>
</div>,
        <Navbar
          setContainerView={[Function]}
/>,
        <Sidebar />,
        false,
        <div
          className="card-container-no-favs start"
>
          No Favorites To Display
</div>,
        <Unknown
          favoriteItem={[Function]}
          selectedGroup={Array []}
/>,
        false,
        <Unknown
          numberOfFavorites={0}
          setContainerView={[Function]}
/>,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1
            className="header-title"
>
            Swapi-Box
</h1>,
          "className": "header",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Swapi-Box",
            "className": "header-title",
          },
          "ref": null,
          "rendered": "Swapi-Box",
          "type": "h1",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "setContainerView": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "No Favorites To Display",
          "className": "card-container-no-favs start",
        },
        "ref": null,
        "rendered": "No Favorites To Display",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "favoriteItem": [Function],
          "selectedGroup": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "numberOfFavorites": 0,
          "setContainerView": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="header"
>
            <h1
                        className="header-title"
            >
                        Swapi-Box
            </h1>
</div>,
          <Navbar
            setContainerView={[Function]}
/>,
          <Sidebar />,
          false,
          <div
            className="card-container-no-favs start"
>
            No Favorites To Display
</div>,
          <Unknown
            favoriteItem={[Function]}
            selectedGroup={Array []}
/>,
          false,
          <Unknown
            numberOfFavorites={0}
            setContainerView={[Function]}
/>,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1
              className="header-title"
>
              Swapi-Box
</h1>,
            "className": "header",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Swapi-Box",
              "className": "header-title",
            },
            "ref": null,
            "rendered": "Swapi-Box",
            "type": "h1",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "setContainerView": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "No Favorites To Display",
            "className": "card-container-no-favs start",
          },
          "ref": null,
          "rendered": "No Favorites To Display",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "favoriteItem": [Function],
            "selectedGroup": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "numberOfFavorites": 0,
            "setContainerView": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
